pkg/pbq/pbq.go

```diff
@@ -46,8 +47,8 @@ var _ ReadWriteCloser = (*PBQ)(nil)
 func (p *PBQ) Write(ctx context.Context, message *isb.ReadMessage) error {
 	// if cob we should return
 	if p.cob {
-		p.log.Errorw("Failed to write message to pbq, pbq is closed", zap.Any("ID", p.PartitionID), zap.Any("header", message.Header))
-		return ErrCOB
+		p.log.Errorw("Failed to write message to pbq, pbq is closed", zap.Any("ID", p.PartitionID), zap.Any("header", message.Header), zap.Any("message", message))
+		return nil
```

Comment:
This is a bit confusing to me, are expect error in this case? If not, then we should not log a error?

Reply:
this causes an infinite loop from which there is no return; hence we are logging and returning `nil`. This is happening because of out-of-order reads from ISB.